### PR TITLE
fix(FEV-1597): handle non-external text tracks

### PR DESCRIPTION
**Pull request by semarche-kaltura**
1. handle loading error;
2. internalisation;
3. init plugin on `TRACKS_CHANGED` event that dispatches for both external and non-external text tracks

### CODE REVIEW

File: src/transcript-plugin.tsx [hunk 1]
```diff
@@ -78,27 +81,39 @@ export class TranscriptPlugin extends KalturaPlayer.core.BasePlugin {
...
+        this._initLoading();
       }
     });
-    this.eventManager.listen(this.player, this.player.Event.TIMED_METADATA_CHANGE, this._onTimedMetadataChange);
-    this.eventManager.listen(this.player, this.player.Event.TIMED_METADATA_ADDED, this._onTimedMetadataAdded);
-    this.eventManager.listen(this.player, this.player.Event.TEXT_TRACK_CHANGED, this._handleLanguageChange);
   }
 
+  private _initLoading = () => {
+    clearTimeout(this._loadingTimeoutId);
```
Comment (lianbenjamin) [inline src/transcript-plugin.tsx:98]: is it possible for `_loadingTimeoutId` to be null or undefined at this point? if so, how would it affect clearTimeout?

**Review by lianbenjamin — CHANGES_REQUESTED** [review]

File: src/transcript-plugin.tsx (hunk 1 above)
Comment (semarche-kaltura) [reply src/transcript-plugin.tsx:98]: check in console
`clearTimeout(null);
clearTimeout(undefined);
clearTimeout(() => {});
clearTimeout();`
you'll see that param can have any type and not required.

Comment (lianbenjamin) [reply src/transcript-plugin.tsx:98]: cool, thanks

File: src/transcript-plugin.tsx [hunk 2]
```diff
@@ -221,6 +241,9 @@ export class TranscriptPlugin extends KalturaPlayer.core.BasePlugin {
   private _addTranscriptItem(): void {
     const {expandMode, position, expandOnFirstPlay} = this.config;
     const {showTime, scrollOffset, searchDebounceTimeout, searchNextPrevDebounceTimeout} = this.config;
+    if (this._transcriptPanel > 0) {
```
[the remark this answers is not in the document]
Comment (semarche-kaltura) [reply src/transcript-plugin.tsx:242]: addressed

**Review by lianbenjamin — APPROVED** [review]

**Pull request merged**